cmd/publishing-bot/publisher.go

```diff
@@ -244,25 +263,48 @@ func (p *PublisherMunger) publish() error {
 }
 
 // Run constructs the repos and pushes them.
-func (p *PublisherMunger) Run() (string, string, error) {
+func (p *PublisherMunger) Run() (string, map[string]string, error) {
 	buf := bytes.NewBuffer(nil)
 	p.plog = NewPublisherLog(buf)
 
-	hash, err := p.updateSourceRepo()
-	if err != nil {
-		p.plog.Errorf("%v", err)
-		p.plog.Flush()
-		return p.plog.Logs(), hash, err
+	// If --branches is specified, re-run the bot with different set of rules
+	// per each specified branch.
+	sourceBranches := strings.Split(p.config.RuleSourceBranches, ",")
+
+	// Each source branch will report its healthz
+	hashes := make(map[string]string, len(sourceBranches))
+	for _, branch := range sourceBranches {
+		hashes[branch] = ""
 	}
-	if err := p.construct(); err != nil {
-		p.plog.Errorf("%v", err)
-		p.plog.Flush()
-		return p.plog.Logs(), hash, err
+
+	var errors []error
+	for _, sourceBranch := range sourceBranches {
+		hash, err := p.updateSourceRepo(sourceBranch)
```

Comment:
Am not convinced to do a "git fetch" here inside the loop. This could lead to inconsistencies if the upstream repo changes in between.